recovery/src/main/java/com/kin/ecosystem/recovery/backup/presenter/BackupPresenterImpl.java

```diff
@@ -2,27 +2,38 @@
 
 
 import android.os.Bundle;
+import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import com.kin.ecosystem.recovery.backup.view.BackupView;
 import com.kin.ecosystem.recovery.base.BasePresenterImpl;
 import com.kin.ecosystem.recovery.events.CallbackManager;
 
 public class BackupPresenterImpl extends BasePresenterImpl<BackupView> implements BackupPresenter {
 
+	private static final String KEY_STEP = "kinrecovery_step";
+	public static final String KEY_ACCOUNT_KEY = "kinrecovery_account_key";
+
 	private @Step
-	int step = STEP_START;
+	int step;
 	private final CallbackManager callbackManager;
 	private boolean isBackupSucceed = false;
+	private final Bundle savedInstanceState;
```

Comment:
I will have property for KEY_STEP and KEY_ACCOUNT_KEY and not savedInstanceState.
in BackupPresenterImpl if savedInstanceState != null extract both. In onSaveInstanceState save both params in outState

Reply:
same as you did on SaveAndSharePresenterImpl